Sources/Purchasing/StoreKitAbstractions/StoreProduct.swift

```diff
@@ -22,6 +23,47 @@ public typealias SK1Product = SKProduct
 @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
 public typealias SK2Product = StoreKit.Product
 
+/// Protocol describing an object with StoreKit.Product's purchase() functions. Helpful
+/// for writing test mocks.
+///
+/// Matches the product.purchase functions in StoreKit 2.
+@available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+internal protocol PurchasableSK2Product: Sendable {
```

Comment:
In order to write tests to verify that we're calling the correct purchase() functions under the correct conditions, I needed a way to write a mock capable of verifying when the purchase functions were called. 

Of course, we can't mock `StoreKit.Product` directly, so instead, I created this `PurchasableSK2Product` protocol that perfectly mirrors `StoreKit.Product`'s purchase functions, and declared an extension to `StoreKit.Product` that conforms to `PurchasableSK2Product`.

Reply:
This could leave us exposed to compilation errors if Apple ever breaks the product.purchase() functions' signatures in a future iOS version, but then again, we and all other SK2 consumers are already vulnerable to that as it is 😅